src/core/symbology/qgspainterswapper.h

```diff
@@ -0,0 +1,52 @@
+/***************************************************************************
+                          qgspainterswapper.h
+                          -------------------
+    begin                : July 2019
+    copyright            : (C) 2019 Hugo Mercier
+    email                : hugo dot mercier at oslandia dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSPAINTERSWAPPER_H
+#define QGSPAINTERSWAPPER_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+
+class QgsRenderContext;
+class QPainter;
+
+/**
+ * \ingroup core
+ * \class QgsPainterSwapper
+ * \brief A class to manage painter saving and restoring required for drawing on a different painter (mask painter for example)
+ *
+ * \since QGIS 3.0
+ */
+class CORE_EXPORT QgsPainterSwapper
+{
+  public:
+
+    /**
+     * QgsPainterSwapper constructor
+     *
+     * \param renderContext the QgsRenderContext object
```

Comment:
Instead of \param approach, I'd suggest:

    The constructor replaces the current painter assigned to \a renderContext, swapping it for the specified \a newPainter. Upon QgsPainterSwapper destruction, the painter previously assigned to the render context will automatically be restored.